.github/workflows/security-check.yml

```diff
@@ -0,0 +1,124 @@
+name: security-check
+
+on:
+  workflow_call:
+    outputs:
+      security_cleared:
+        description: "Whether the PR passed security checks"
+        value: ${{ jobs.security-gate.outputs.security_cleared }}
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+# Set permissions at workflow level
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+
+jobs:
+  security-gate:
+    name: External Contribution Security Review
+    runs-on: ubuntu-latest
+    outputs:
+      security_cleared: ${{ steps.check.outputs.security_cleared }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check for security concerns
+        id: check
+        shell: bash
+        env:
+          PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+          REPO: ${{ github.repository }}
+          REPO_OWNER: ${{ github.repository_owner }}
+          BASE_SHA: ${{ github.event.pull_request.base.sha }}
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
"Curling the curl": could someone just add an `echo ${{ secrets.GITHUB_TOKEN }}` to the script below and get the access credentials? Or do we need to manually run this script for users outside the organization?